cmd/clusterctl/client/repository/components.go

```diff
@@ -292,9 +301,73 @@ func addNamespaceIfMissing(objs []unstructured.Unstructured, targetNamespace str
 	return objs
 }
 
+func fixWebHookTargetNamespace(o unstructured.Unstructured, targetNamespace string) (unstructured.Unstructured, error) {
+	if o.GetKind() == mutatingWebhookConfigurationKind || o.GetKind() == validatingWebhookConfigurationKind {
+		annotations := o.GetAnnotations()
+		secretNamespacedName, ok := annotations["cert-manager.io/inject-ca-from"]
+		if ok {
+			secretName := strings.Split(secretNamespacedName, "/")[1]
+			annotations["cert-manager.io/inject-ca-from"] = targetNamespace + "/" + secretName
+			o.SetAnnotations(annotations)
+		}
+	}
+
+	var err error
+	switch o.GetKind() {
+	case mutatingWebhookConfigurationKind:
+		b := &admissionregistrationv1.MutatingWebhookConfiguration{}
+		if err := scheme.Scheme.Convert(&o, b, nil); err != nil {
+			return o, err
+		}
+
+		whs := []admissionregistrationv1.MutatingWebhook{}
+		for _, w := range b.Webhooks {
+			if w.ClientConfig.Service != nil {
+				w.ClientConfig.Service.Namespace = targetNamespace
+			}
+			whs = append(whs, w)
+		}
+		b.Webhooks = whs
```

Comment:
```suggestion
		for _, w := range b.Webhooks {
			if w.ClientConfig.Service != nil {
				w.ClientConfig.Service.Namespace = targetNamespace
			}
		}
```
That should also work (same for ValidatingWebhook)